### DISCUSSION TITLE
docker error- exited code 1

### DISCUSSION THREAD

**Discussion by shaybit** [ptarmiganlabs/butler-sos#424] [Q&A]
Hi, I followed the documentation as much as I could and encountered error as decribed below:

1. I created the root folder(named butler-sos-docker) and its subfolders as described here : https://butler-sos.ptarmiganlabs.com/docs/getting_started/install/docker/
2. downloaded the config file from https://github.com/ptarmiganlabs/butler-sos/blob/master/src/config/production_template.yaml, named it production.yaml and moved it to butler-sos-docker/config/
3. Edited the config file and filled the necessary according the  page : https://butler-sos.ptarmiganlabs.com/docs/getting_started/setup/
4. Downloaded the docker file from : https://github.com/ptarmiganlabs/butler-sos/blob/master/docs/docker-compose/docker-compose_fullstack_influxdb.yml and named it docker-compose.yml and  changed only the NODE_ENV variable to be "NODE_ENV=production" since my config file name is productio
5. Added the qlik certificates to the butler-sos-docker/config/certificate folder
6. Enter the terminal to the root folder location and ran docker-compose up then I got the following error

butler-sos    | /nodeapp/node_modules/influx/lib/src/schema.js:83
butler-sos    |             throw new Error("Extraneous tags detected for writing InfluxDB point in " +
butler-sos    |                   ^
butler-sos    |
butler-sos    | Error: Extraneous tags detected for writing InfluxDB point in SenseOps.sense_server: `server-type`.
butler-sos    |     at Schema.checkTags (/nodeapp/node_modules/influx/lib/src/schema.js:83:19)
butler-sos    |     at InfluxDB.parsePoint (/nodeapp/node_modules/influx/lib/src/index.js:919:43)
butler-sos    |     at /nodeapp/node_modules/influx/lib/src/index.js:834:89
butler-sos    |     at Array.forEach (<anonymous>)
butler-sos    |     at InfluxDB.writePoints (/nodeapp/node_modules/influx/lib/src/index.js:833:16)
butler-sos    |     at Object.postHealthMetricsToInfluxdb (/nodeapp/src/lib/post-to-influxdb.js:208:10)
butler-sos    |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
butler-sos    |
butler-sos    | Node.js v18.2.0
butler-sos exited with code 1

the docker ps command shows:

CONTAINER ID   IMAGE                             COMMAND                  CREATED          STATUS                             PORTS                                            NAMES
d5a759d3c7ac   ptarmiganlabs/butler-sos:latest   "docker-entrypoint.s…"   13 minutes ago   Up 20 seconds (health: starting)                                                    butler-sos
9ff371bbc1dc   grafana/grafana:latest            "/run.sh"                13 minutes ago   Up 20 seconds                      0.0.0.0:3000->3000/tcp                           grafana
a835a28b59d5   influxdb:1.8.10                   "/entrypoint.sh infl…"   13 minutes ago   Up 20 seconds                      0.0.0.0:8082->8082/tcp, 0.0.0.0:8086->8086/tcp   influxdb

Thanks for your assistance

**Comment by mountaindude** [comment]
The error message suggests a problem around Butler SOS storing data in InfluxDB.
I believe `Extraneous tags detected for writing InfluxDB point ` can occur if are trying to write Butler SOS data to an existing InfluxDB database that has a different schema/data model than Butler SOS expects.

This could happen also if you modify the tags used (e.g. `server-type` in this case).
For example, if you have been running Butler SOS for a while but then decide to add or remove a tag, this could cause problems as InfluxDB is pretty strict when it comes to its schema.

You could try deleting the Butler SOS database in InfluxDB. I.e. start over.
Butler SOS will create a new database next time it starts. 
The downside is that you will loose whatever Butler SOS data you have in that database, but given that Butler SOS deals with operational data, you probably shouldn't keep more than a week or two of historical records in that database. 
Usually fine to delete - but that's for you to decide.



This is a fairly well tested use case but we could also be looking at a bug, of course.
I will test things here and see if I can reproduce the problem.

**Comment by mountaindude** [comment]
Looks like there is a minor - but important - typo in the sample config file.
Most users tend to just delete those sample tags as they are just examples, but as examples they should of course work.
I'll update the sample config file in the GitHub repository.

I suspect this is what's causing your troubles, but we can't be sure until you have tested of course.
Give it a try and report back here and we'll take it from there.

<img width="1314" alt="image" src="https://user-images.githubusercontent.com/1029262/184074187-2a5022d1-da7a-430f-aeb4-49700c4ebb4c.png">
